File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.java
class Solution {
    public long largestPerimeter(int[] nums) {
        
        
         Arrays.sort(nums);
        int n = nums.length;
        long [] pref = new long[n];
        
        pref[0] = nums[0];
        for (int i = 1; i < n; i++) {
            pref[i] = pref[i - 1] + nums[i];
        }
        int ans = -1;
        for (int i = n - 1; i >= 2; i--) {
            int longest = nums[i];
            if (pref[i - 1] > longest) {
                return pref[i - 1] + longest;
            }
        }
        return ans;
        
    }
    
    
}

File: 3262-find-polygon-with-the-largest-perimeter/find-polygon-with-the-largest-perimeter.py
'''
solution pref sum + greedy


the idea is we sort the  nums array 
and we create pref array for this nums

pref[i]which denote sum of all sides from o to i 

next we iterate on nums from end , checking if nums[i] can be the longest 
in which case pref sum till i-1 should be> nums[i]

t n 

s n 
'''
#python
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        pref = [0 for i in range(n )]
        
        pref[0] = nums[0]
        for i in range(1,n):
            pref[i] = pref[i-1]+ nums[i]
        ans = -1
        j = n-2
        for i in range(n-1,1,-1):
            
            longest = nums[i]
  

            if pref[i-1]>longest:
                return pref[i-1]+ longest 
        return ans
            